src/lib/features/feature-search/feature-search-store.ts

```diff
@@ -371,6 +378,53 @@ class FeatureSearchStore implements IFeatureSearchStore {
                     },
                 )
                 .select('feature_cr.change_request_ids');
+
+            finalQuery
```

Comment:
This could be a view or extracted method but the query is here to stay

Reply:
I think extracting the method is a good direction. We do not have many of that currently, but if we can have these parts as separate methods, the method naming can give good indication what is being done in each joins.